### PR TITLE
[react, react-core] - rely on react-core in react

### PR DESCRIPTION
**Pull request by jnsdls**
Dashboard with this version of `react` package:
https://thirdweb-www-git-jd-use-new-react-core.thirdweb-preview.com/

### CODE REVIEW

**Review by ikethirdweb — CHANGES_REQUESTED** [review]
looking great! Some comments before approving.

**Comment by ikethirdweb** [comment]
I think we can also remove `react-core/src/evm/constants/ipfs.ts` and `react-core/src/evm/constants/runtime.ts`. 

As for the file: `react-core/src/evm/constants/chain.ts`, we can leave it in `react-core`. Note that I extracted the `Chain` type into `react-core/src/evm/types.ts`.

File: packages/react-core/package.json [hunk 1]
```diff
@@ -80,10 +80,10 @@
   },
   "dependencies": {
     "@tanstack/react-query": "^4.0.10",
-    "buffer": "^6.0.3",
     "tiny-invariant": "^1.2.0"
   },
   "peerDependencies": {
+    "@solana/wallet-adapter-react": "^0.15.19",
```
[the remark this answers is not in the document]
Comment (jnsdls) [reply packages/react-core/package.json:86]: the "wallet" type that gets passed from this is similar to the "signer" in evm, so we kind of need it for now I believe

File: packages/react-core/src/solana/providers/base.tsx [hunk 2]
```diff
@@ -8,12 +8,14 @@ import {
...
   ThirdwebAuthConfigProvider,
 } from "../contexts/thirdweb-auth";
+import type { WalletContextState } from "@solana/wallet-adapter-react";
 import { Network, ThirdwebSDK } from "@thirdweb-dev/sdk/solana";
 import { createContext, useContext, useEffect, useMemo, useState } from "react";
 import invariant from "tiny-invariant";
 
 interface ThirdwebSDKProviderProps extends QueryClientProviderProps {
   network: RequiredParam<Network>;
+  wallet?: WalletContextState;
```
[the remark this answers is not in the document]
Comment (jnsdls) [reply packages/react-core/src/solana/providers/base.tsx:18]: good catch!

**Comment by jnsdls** [comment]
> I think we can also remove `react-core/src/evm/constants/ipfs.ts` and `react-core/src/evm/constants/runtime.ts`.
> 
> As for the file: `react-core/src/evm/constants/chain.ts`, we can leave it in `react-core`. Note that I extracted the `Chain` type into `react-core/src/evm/types.ts`.

`runtime` is still used in `react-core` so keeping that but good call on ipfs

**Review by ikethirdweb — APPROVED** [review]
- Please, remove `constants/ipfs` from `src/evm/index.ts` before merging

**Pull request merged**